cylc/rose/entry_points.py

```diff
@@ -0,0 +1,207 @@
+
+        # Process files
+        config_pm = ConfigProcessorsManager(event_handler, popen, fs_util)
+        config_pm(config_tree, "file")
+
+    return True
+
+
+def record_cylc_install_options(
+    dest_root=None,
+    opts=None,
+    dir_=None,
+):
+    """Create/modify files recording Cylc install config options.
+
+    Creates a new config based on CLI options and writes it to the workflow
+    install location as ``rose-suite-cylc-install.conf``. If
+    ``rose-suite-cylc-install.conf`` already exists over-writes changed items,
+    except for ``!opts=`` which is merged and simplified.
+
+    If ``!opts=`` have been changed these are appended to those that have
+    been written in the installed ``rose-suite.conf``.
+
+    Args:
+        _ (pathlib.Path | or str):
+            Not used in this function, but requried for consistent entry point.
+        opts:
+            Cylc option parser object - we want to extract the following
+            values:
+            - opt_conf_keys (list or str):
+                Equivelent of ``rose suite-run --option KEY``
+            - defines (list of str):
+                Equivelent of ``rose suite-run --define KEY=VAL``
+            - suite_defines (list of str):
+                Equivelent of ``rose suite-run --define-suite KEY=VAL``
+        dest_root (pathlib.Path | or str):
+            Path to dump the rose-suite-cylc-conf
+
+    Returns:
+        cli_config - Config Node which has been dumped to
+        ``rose-suite-cylc-install.conf``.
+        rose_suite_conf['opts'] - Opts section of the config node dumped to
+        installed ``rose-suite.conf``.
+    """
+    # Construct a path objects representing our target files.
+    conf_filepath = Path(dest_root) / 'rose-suite-cylc-install.conf'
+    rose_conf_filepath = Path(dest_root) / 'rose-suite.conf'
+    dumper = ConfigDumper()
+    loader = ConfigLoader()
+
+    # Create a config based on command line options:
+    cli_config = get_cli_opts_node(opts)
+
+    # If file exists we need to merge with our new config, over-writing with
+    # new items where there are duplicates.
+
+    if conf_filepath.is_file():
+        oldconfig = loader.load(str(conf_filepath))
+        cli_config = merge_rose_cylc_suite_install_conf(oldconfig, cli_config)
+    dumper(cli_config, str(conf_filepath))
+
+    cli_config.comments = [
+        ' This file records CLI Options.'
+    ]
+    dumper.dump(cli_config, str(conf_filepath))
+
+    # Replace the opts section of the rose-suite.conf in the install location.
+    # If we have not a rose-suite.conf but we use one of the Rose-style
+    # options we still want to record those options.
+    if not rose_conf_filepath.is_file():
+        rose_conf_filepath.touch()
+    rose_suite_conf = loader.load(str(rose_conf_filepath))
+    rose_suite_conf = get_installed_rose_suite_conf_node(
+        rose_suite_conf, cli_config
+    )
+    dumper(rose_suite_conf, rose_conf_filepath)
```

Comment:
Need to fiddle the `opts` value to insert the `cylc-install` optional config (if one was written) otherwise subsequent Rose commands will not load it.

Reply:
See my comments about misunderstanding how this is meant to work.